### ISSUE TITLE
Add caching for script properties and engine metadata requests

### ISSUE THREAD

**Issue by Rikonardo** [CheshireCaat/bas-remote-node#24]
## Problem

I've experienced an issue when using a lot of simultaneous browser instances in short sessions (under 60 seconds) - after some time launch method started failing with json parsing error:
![image psd](https://github.com/user-attachments/assets/69959006-931c-42a6-8dc8-65631fa5c396)
I've traced this error back to request that happen before every browser launch - fetching script properties. Manually checking URL in browser confirms it - in 80-90% of requests, 503 error is returned:
![image psd(1)](https://github.com/user-attachments/assets/43c14022-fb96-4070-95fa-765e11ad76ac)

While this doesn't really look like an intentional rate limiting, this error is definitely tied to amount of requests made from single IP, as it started happening only after increasing worker count. I also tested it on my second IP, and error started appearing after manually sending large amount of requests. 
The second link (engine metadata) doesn't exhibit this issue, but it makes sense to include it as well, as those requests are tightly related in code.

As there is no caching of this data, in a use case where browser reruns often, we are effectively checking for updates every minute, and, considering there are dozens of workers running at the same time, spamming bablosoft servers with multiple requests per second, which eventually leads to 503 error on script properties endpoint.

## Proposed solution

Implement a (possibly optional, preferably on an opt-out basis) caching mechanism for those requests.

E.g. `EngineService.initilize()` is called with `options.scriptName = "FingerprintPluginV9"` 
Before doing any requests, we read `${_scriptDir}/.properties-cache.x${ARCH}.json` (e.g. `"engine/run/FingerprintPluginV9/.properties-cache.x64.json"`), which contains timestamp and responses to both initial requests. If time passed is less than a certain threshold (e.g. 10 minutes), we use file contents instead of sending requests.
If there were no file, or cache is expired, we do requests and store responses as cache.

I can draft a PR if you're okay with this idea. 

I really hope you'll respond and address this quickly, as this issue makes it impossible for me to run large-scale operations from a single network.
Ping: @CheshireCaat

**Comment by CheshireCaat** [comment]
Greetings, your idea sounds good.

Can be added as an optional setting for the client:

```js
const client = new BasClient({
  enablePropertiesCache: true,
});
```

Let it be disabled by default - in order to avoid problems with other projects.
Later I'll enable this setting or add a method for configuration on the automation plugins side.

In the future, I think it's worth moving the cache to some more common folder - in order to avoid such an error when clients run from different working directories, but use the same project.

I will be glad to see your PR.

**Comment by CheshireCaat** [comment]
@Rikonardo hi, in the near future, I will release a version with caching functionality myself, I am working on it at the moment.

**Comment by Rikonardo** [comment]
Oh, that's great news! Past days were pretty busy, so I wasn't able to finish my implementation. I've got kinda stuck on finding a better way to pass this option into a client, because I need this to work with `playwright-with-fingerprints` (and thus `browser-with-fingerprints`, which uses a static instance of client)

**Comment by CheshireCaat** [comment]
I decided to use [file-system-cache](https://www.npmjs.com/package/file-system-cache) with 10 minutes ttl. Seems to enough for me, later we can add ability to customize options, but I think it doesn't really matter right now.

Also the common folder inisde user **temp** directory will be used for cache, because problems with requests can occur, including when working with different scripts and with different processes.

I will publish an update today.

**Comment by CheshireCaat** [comment]
Fixed in 90455d1
